app/models/loader/orchestrate.rb

```diff
@@ -118,6 +118,67 @@ def load
       store_passwords
 
       store_public_keys
+
+      emit_audit
```

Comment:
We need to emit these after the transaction is committed, and keep track of whether each one has been sent.

Reply:
I was thinking about it and I really don't think it's necessary. If audit can't be sent in full, the transaction should just be rolled back.

Reply:
But then you have audits of some things happening that never happened?

Reply:
Only in tight disaster scenarios, which I doubt will ever happen in practice. Even if it does happen, the entries be easily identified as spurious by virtue of duplicate policy version (or referring to a nonexistent one).

Reply:
What happens if the process reading data from the `audit_socket` crashes in the middle of reading events from the socket? Will that cause the transaction to abort? If so, and given that we don't have control over that process, I can definitely imagine that happening in practice.

OTOH, we're storing the audit records in the database, right? So, I guess we could say that the db table is the system of record, and the audit stream might have spurious events in it.

Reply:
It will. But we do have control over that process, at least on the appliance; namely, it's the local syslog daemon. It's highly unlikely it crashes in the middle of reading events from the socket.

Reply:
We won't have control over other processes, but this is the responsibility of the user. Conjur core does not provide a system of record for the audit data, it just produces an event stream.